fe/fe-core/src/main/java/org/apache/doris/analysis/StmtRewriter.java

```diff
@@ -1141,5 +1154,52 @@ private static Expr createJoinConjunct(Expr exprWithSubquery, InlineViewRef inli
         smap.put(subquery, subquerySubstitute);
         return exprWithSubquery.substitute(smap, analyzer, false);
     }
+
+    public static boolean rewriteByPolicy(StatementBase statementBase, Analyzer analyzer) throws UserException {
+        Catalog currentCatalog = Catalog.getCurrentCatalog();
+        if (!(statementBase instanceof SelectStmt)) {
+            return false;
+        }
+        SelectStmt selectStmt = (SelectStmt) statementBase;
+        boolean reAnalyze = false;
+        for (int i = 0; i < selectStmt.fromClause_.size(); i++) {
+            TableRef tableRef = selectStmt.fromClause_.get(i);
+            // Recursively rewrite subquery
+            if (tableRef instanceof InlineViewRef) {
+                InlineViewRef viewRef = (InlineViewRef) tableRef;
+                if (rewriteByPolicy(viewRef.getQueryStmt(), analyzer)) {
+                    reAnalyze = true;
+                }
+                continue;
+            }
+            Table table = tableRef.getTable();
+            String dbName = tableRef.getName().getDb();
+            if (dbName == null) {
+                dbName = analyzer.getDefaultDb();
+            }
+            Database db = currentCatalog.getDbOrAnalysisException(dbName);
+            long dbId = db.getId();
+            long tableId = table.getId();
+            String user = analyzer.getQualifiedUser();
+            Policy matchPolicy = currentCatalog.getPolicyMgr().getMatchRowPolicy(dbId, tableId, user);
```

Comment:
I think we can move this check in front of the method, so that if there is no policy at all, the method can be returned directly.